raiden-ts/tests/unit/utils.spec.ts

```diff
@@ -328,7 +329,8 @@ describe('data', () => {
     expect(() => encode(-1, 2)).toThrowError('negative');
     expect(() => encode(bigNumberify(65537), 2)).toThrowError('too large');
     expect(() => encode('0x01', 2)).toThrowError(ErrorCodes.DTA_ARRAY_LENGTH_DIFFRENCE);
-    expect(() => encode((true as unknown) as number, 2)).toThrowError('data is not');
+    // @ts-ignore // TODO: replace with @ts-expect-error ts@^3.9
+    expect(() => encode(null, 2)).toThrowError('data is not');
```

Comment:
Could you explain the commend above, also I am trying to get the `data is not` will that reach the SDK user at some point?  Looking this my question would be "data is not what?"

Reply:
This function accepts some types and encode them to serialized forms of given size. This test ensures that, when passed a type that isn't handled, it throws. Before, it was boolean; now I handle boolean and therefore had to change the test to something else, i.e. `null`.

To test that, I need to pass a type which isn't accepted by the function declaration. As TS would complain, I need to ts-ignore (which is bad). On TypeScript 3.9 beta, there's a new declaration specifically for this kind of usecase: `// @ts-expect-error`, which makes TS accept something wrong from its perspective (useful for unit tests), but differently from ts-ignore, it'll error if there's no compile-time error on that line, so you get a more strict validation of errors. E.g. if in the future we start accepting `null`, `@ts-expect-error` would complain, while `@ts-ignore` would just ignore; of course in this case, the test would fail as well, but it's useful to have proper typing at development time as well.

The message thrown is variable and dependent of the input type, that's why I just include a substring of it here, just to be notified if the message changes for some reason (e.g. it becomes a `TypeError` due to some change in the encoding algorithm while I expected this specific string error). I could put some more specific part of it there, but I didn't think it was worth it, but I'll change that.

Reply:
Anyway, that error should not reach users at any point, it's something for internal helper/utility function, more of an assert.

Reply:
Ok thanks for clarifying, if it is only internal then no need to change :)